src/main/java/com/yelp/nrtsearch/server/luceneserver/state/BackendGlobalState.java

```diff
@@ -202,6 +225,10 @@ public void setStateDir(Path source) throws IOException {
     throw new UnsupportedOperationException();
   }
 
+  private void setImmutableState(ImmutableState immutableState) {
+    this.immutableState = immutableState;
+  }
```

Comment:
nit: as this is a separate method, it might be good to verify `Thread.holdsLock(this)` before updating the active state.

Reply:
Should I make this method synchronized or just remove this method but just add this line in the synchronized reloadStateFromBackend method directly?

Reply:
I would probably either remove it, or throw an exception if the thread does not hold the object monitor.